quinn-proto/src/connection/streams.rs

```diff
@@ -504,8 +506,13 @@ impl Streams {
             if rs.is_finished() {
                 continue;
             }
+
+            let max = match rs.window_update(self.stream_receive_window, false) {
+                Some(max) => max,
+                None => continue,
+            };
```

Comment:
If this packet is lost, this change will lead us to fail to retransmit the `MAX_STREAM_DATA` frame, possibly deadlocking the sender. We should instead perform this check before adding the stream to `pending.max_stream_data` and leave frame construction unchanged, ensuring that a loss will cause the latest data to be retransmitted.

Reply:
Oh, that's not good. I thought Quinn retransmits all lost frames completely, and therefore thought this is good. 

So the idea is: Anytime `pending` is set it grabs the highest possible window from the stream. And the `read` methods make sure pending is only set once whenever the increment is high enough? I think this has a catch, but I will look into it.

Reply:
> I thought Quinn retransmits all lost frames completely

We keep track of what needs to be retransmitted if a packet is deemed lost by recording the retransmittable frames carried by each in-flight packet, and re-queueing those frames when the packet is deemed lost. See `sent.max_stream_data.insert(id)` below. The key is that that data's abstract; we don't store the literal frames, because they might become out of date. Instead, we just rerun this exact code again when it's time to retransmit.

> So the idea is:

Any time a stream appears in `pending.max_stream_data`, yeah, that's the idea exactly.

Reply:
I think the new logic should cover it. The transmission will now suppress further updates, but retransmits can still grab the value